packages/server/graphql/public/mutations/upsertTeamPromptResponse.ts

```diff
@@ -94,8 +94,10 @@ const upsertTeamPromptResponse: MutationResolvers['upsertTeamPromptResponse'] =
     publish(
       SubscriptionChannel.NOTIFICATION,
       notification.userId,
-      'UpsertTeamPromptResponseSuccess',
-      data,
+      'AddedNotification',
+      {
+        addedNotificationId: notification.id
+      },
       subOptions
     )
```

Comment:
This could be abstracted into something like `publishNotification(notification, subOptions)`.

Reply:
i like it! waaay long ago we used to use a similar pattern for all entities (e.g. AddedTeam, UpdatedTeam, RemovedTeam) it worked really well! Eventually we had to switch because the client side logic sometimes needed to know which event triggered the change, which fields specifically were updated, etc.

I think something like this where it's only handling the C in CRUD & only handling the Notification entity is a great compromise on simplicity & flexibility. The question we have to ask ourselves is, "If 2 different mutations can create the same notification, will the client care which mutation caused it?" For example, if someone gets invited to a team via a push invitation, does the client treat it differently vs. if another mutation triggered the invited to team notification? If that case arises, is there a workaround (e.g. fall back to using the SuccessPayload?)

tl;dr if you can't think of any cases where the client needs to know the mutation that triggered it, it's probably a safe pattern!

Reply:
I think the way most notifications get loaded generally means that the client can't care which mutation triggered it - when notifications are loaded on page load, we just load the notification object with no context into where the notification came from.

There _may_ be certain exceptions to this for e.g. team invitations or org removals, but those are less "hey, just letting you know about this thing!" and more "something very specific just changed and we need to change some state". For notifications that only need us to add the notification to the notifications drawer (and maybe pop a toast based on the notification type), this should work fine.

Reply:
totally agree!